types/d3-array/d3-array-tests.ts

```diff
@@ -784,48 +784,1584 @@ mergedArray = d3Array.merge([testArray1, [15, 30]]); // fails, type mismatch
+                                                                                                                            ]
+                                                                                                                        >,
+                                                                                                                    ]
+                                                                                                                >,
+                                                                                                            ]
+                                                                                                        >,
+                                                                                                    ]
+                                                                                                >,
+                                                                                            ]
+                                                                                        >,
+                                                                                    ]
+                                                                                >,
+                                                                            ]
+                                                                        >,
+                                                                    ]
+                                                                >,
+                                                            ]
+                                                        >,
+                                                    ]
+                                                >,
+                                            ]
+                                        >,
+                                    ]
+                                >,
+                            ]
+                        >,
+                    ]
+                >,
+            ]
+        >,
+    ]
+> = d3Array.groups(
+    objArray,
+    d => d.role,
+    d => d.name,
+    d => d.date,
+    (d, index) => index % 2,
+    (d, index, values) => values.length,
+    d => d.role,
+    d => d.name,
+    d => d.date,
+    (d, index) => index % 2,
+    (d, index, values) => values.length,
+    d => d.role,
+    d => d.name,
+    d => d.date,
+    (d, index) => index % 2,
+    (d, index, values) => values.length,
+    d => d.role,
+    d => d.name,
+    d => d.date,
+    (d, index) => index % 2,
+    (d, index, values) => values.length,
+);
+
+// indexes() ---------------------------------------------------------------------
+
 const indexes: Array<[string, ObjDefinition]> = d3Array.indexes(objArray, d => d.name);
-const indexes2: Array<[string, Array<[string, ObjDefinition]>]> = d3Array.indexes(objArray, d => d.name, d => d.date);
```

Comment:
Reviewing this would be a bit easier if this change didn't include a load of unrelated formatting changes; I do see what's happening in the index and it's probably fine, but the tests really have these looking huge and different.